subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -76,6 +77,8 @@
 
 public class HttpClientConfigurer {
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientConfigurer.class);
+    private final int connectionTimeout = 10000;
```

Comment:
:x: It seems that `connectionTimeout` and `socketTimeout` are constants. I'd either make them proper constants, or make them configurable. Making them configurable might help with testing.